cmd/controller-manager/duplicate_detector.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	ctrlutil "github.com/kubestellar/kubestellar/pkg/ctrlutil"
+	kfv1alpha1 "github.com/kubestellar/kubeflex/api/v1alpha1"
+	kslclient "github.com/kubestellar/kubestellar/pkg/client"
+)
+
+type ObjectIdentifier struct {
+	Group     string
+	Version   string
+	Kind      string
+	Name      string
+	Namespace string
+}
+
+// DetectDuplicateObjectIdentifiersAcrossWDSes checks for duplicate object identifiers across all WDSes and logs warnings if found.
+func DetectDuplicateObjectIdentifiersAcrossWDSes(logger logr.Logger) error {
+	ctx := context.Background()
+	kubeClient := *kslclient.GetClient()
+	labelSelector := labels.SelectorFromSet(labels.Set(map[string]string{
+		ctrlutil.ControlPlaneTypeLabel: ctrlutil.ControlPlaneTypeWDS,
+	}))
+	wdsList := &kfv1alpha1.ControlPlaneList{}
+	if err := kubeClient.List(ctx, wdsList, &client.ListOptions{LabelSelector: labelSelector}); err != nil {
+		return fmt.Errorf("failed to list WDS control planes: %w", err)
+	}
+
+	idToWDS := map[ObjectIdentifier][]string{}
+
+	for _, wds := range wdsList.Items {
+		wdsName := wds.Name
+		restConfig, _, err := ctrlutil.GetWDSKubeconfig(logger, wdsName)
+		if err != nil {
+			logger.Error(err, "Failed to get kubeconfig for WDS", "wds", wdsName)
+			continue
+		}
+		dynClient, err := dynamic.NewForConfig(restConfig)
```

Comment:
Please learn from the code in the Binding controller that also enumerates all the workload objects in a WDS.

Rather than have two copies of this code, it would be much better to refactor to a DRY solution.